presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java

```diff
@@ -0,0 +1,163 @@
+
+    private static final String START_METHOD =
+            "execution(* io.prestosql.dispatcher.DispatchInfo"
+                    + ".getQueuedTime(..))";
+
+    private static final String WHITE_AND_BLACK_LIST =
+            "execution(* io.prestosql..*(..)) && " +
+                    "!within(io.prestosql.logging..*)";
+
+    private static final String FINISH_METHOD =
+            "execution(* io.prestosql.execution.TaskInfo.isNeedsPlan())";
+
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * This method is executed when the Thread starts execution of creation schema command inside
+     * worker.
+     *
+     * <p>It is necessary to create the step and debug log id variable for each thread that will
+     * execute the flux. So it will delegate the processing to {@link
+     * BaseAspect#printDebugLogForMethod(ProceedingJoinPoint, long)} that will be responsible to
+     * insert inside the log information about method to be executed.
+     *
+     * @param point an object o Aspectj library that represents the method that ShannonDB needs to
+     * obtain information inside logs.
+     * @return the same object that is returned by the method that is wrapped in this advice
+     * @throws Throwable the same exception that is thrown by the method that is wrapped
+     */
+    @Around(START_METHOD)
+    public Object startFlux(final ProceedingJoinPoint point)
+            throws Throwable
+    {
+        final long threadId = Thread.currentThread().getId();
+
+        threadIdToStep.put(threadId, 0);
+        threadIdToDebugLogId.compute(
+                threadId, (key, value) -> UUID.randomUUID().getMostSignificantBits());
+
+        return printDebugLogForMethod(point, threadId);
+    }
+
+    /**
+     * This method just retrieves the thread id and delegate the processing to method inside {@link
+     * BaseAspect}
+     *
+     * <p>It will delegate the processing to {@link
+     * BaseAspect#printDebugLogForMethod(ProceedingJoinPoint, long)} that will be responsible to
+     * insert inside the log information about method to be executed.
+     *
+     * @param point an object o Aspectj library that represents the method that ShannonDB needs to
+     * obtain information inside logs.
+     * @return the same object that is returned by the method that is wrapped in this advice
+     * @throws Throwable the same exception that is thrown by the method that is wrapped
+     */
+    @Around(WHITE_AND_BLACK_LIST)
+    public Object around(final ProceedingJoinPoint point)
+            throws Throwable
+    {
+        final long threadId = Thread.currentThread().getId();
+
+        return printDebugLogForMethod(point, threadId);
+    }
+
+    /**
+     * This method is executed when the Thread starts execution when finishes the command of create
```

Comment:
I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

Reply:
fixed